File: docs/node-mjs/playwright/html.mjs
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch();
  const context = await browser.newContext();
  const page = await context.newPage();

  const url = 'https://example.com';
  
  // 1️⃣ Get server-rendered HTML (prior to JS execution)
  const response = await page.goto(url, { waitUntil: 'domcontentloaded' });
  const serverHTML = await response.text();
  console.log('--- Server-rendered HTML ---');
  console.log(serverHTML.slice(0, 500), '...');

  // 2️⃣ Let client JS run and get the final DOM
  await page.waitForLoadState('networkidle');
  const clientHTML = await page.content();
  console.log('--- Client-rendered HTML ---');
  console.log(clientHTML.slice(0, 500), '...');

  await browser.close();
})();

File: docs/node-mjs/playwright/html.py
from playwright.async_api import async_playwright

async with async_playwright() as p:
    browser = await p.chromium.launch()
    context = await browser.new_context()
    page = await context.new_page()

    resp = await page.goto("https://example.com", wait_until="domcontentloaded")
    server_html = await resp.text()
    print("Server HTML:", server_html[:500], "...")

    await page.wait_for_load_state("networkidle")
    client_html = await page.content()
    print("Client HTML:", client_html[:500], "...")

    await browser.close()
